index_mh_iter.go

```diff
@@ -1,22 +1,22 @@
-package suppliers
+package provider
 
 import (
 	"context"
 	"io"
 
-	provider "github.com/filecoin-project/indexer-reference-provider"
 	"github.com/ipld/go-car/v2/index"
 	"github.com/multiformats/go-multihash"
 )
 
-var _ provider.MultihashIterator = (*indexMhIterator)(nil)
+var _ MultihashIterator = (*indexMhIterator)(nil)
 
 type indexMhIterator struct {
 	mhch chan multihash.Multihash
 	err  error
 }
 
-func newIndexMhIterator(ctx context.Context, idx index.IterableIndex) *indexMhIterator {
+// NewMultihashIteratorFromIterableIndex constructs a new MultihashIterator from a CAR index.IterableIndex.
```

Comment:
This should document what the context is used for.

Reply:
Done; I'd love your quick eye on the docs before merging please 🍻